projects/keycloak/SamlParserFuzzer.java

```diff
@@ -48,28 +109,227 @@ public static void fuzzerTestOneInput(FuzzedDataProvider data) {
       // instance and run the parse method with the
       // random data provided by the XMLEventReader
       // object created above
-      switch (data.consumeInt(1, 10)) {
+      switch (data.consumeInt(1, 73)) {
         case 1:
-          parser = SAMLParser.getInstance();
+          parser = HttpClientParser.getInstance();
+          break;
```

Comment:
Why do we need `break` at the end of a case? It is redundant.

Reply:
Java case is a label, if it is not break, the logic will continue and run the next case.

Reply:
break is used to immediately exit the switch statement without running the next statement. And case is only indicate the starting point when condition is met, it does not meant that the remaining code will not run.

Reply:
what is "the next statement" here? If case `4` is true and we don't break, we will invoke case `5` as well?

Reply:
```java
Integer a = 3;
switch (a) {
  case 1: System.out.println("A");
  case 2: System.out.println("B");
  case 3: System.out.println("C");
  case 4: System.out.println("D");
  case 5: System.out.println("E");
}
```
The above will print CDE.

```java
Integer a = 3;
switch (a) {
  case 1: System.out.println("A"); break;
  case 2: System.out.println("B"); break;
  case 3: System.out.println("C"); break;
  case 4: System.out.println("D"); break;
  case 5: System.out.println("E"); break;
}
```
The above will print C.

Reply:
Did not know that, thanks for elaborating.